okhttp-tests/src/test/java/okhttp3/internal/tls/CertificatePinnerChainValidationTest.java

```diff
@@ -197,8 +199,20 @@
         .issuedBy(compromisedIntermediateCa)
         .commonName(server.getHostName())
         .build();
-    SslClient serverSslContext = new SslClient.Builder()
-        .certificateChain(rogueCertificate.keyPair, rogueCertificate.certificate, compromisedIntermediateCa.certificate, goodCertificate.certificate, rootCa.certificate)
+
+    SslClient.Builder sslBuilder = new SslClient.Builder();
+
+    // Test setup fails on JDK9
+    // java.security.KeyStoreException: Certificate chain is not valid
+    // at sun.security.pkcs12.PKCS12KeyStore.setKeyEntry
+    // http://openjdk.java.net/jeps/229
+    // http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/2c1c21d11e58/src/share/classes/sun/security/pkcs12/PKCS12KeyStore.java#l596
+    if (getPlatform().equals("jdk9")) {
```

Comment:
Should this workaround just go in SslClient.Builder? Seems simpler and avoids distracting from the tests?